### ISSUE TITLE
Quickly dismissing 3DS VC crashes the app

### ISSUE THREAD

**Issue by nickdnk** [stripe/stripe-ios#1858]
## Summary
Immediately dismissing the 3D Secure view controller (presented by `handleNextAction()`) using the cancel button on the top right crashes the app. You have to be quite quick, but it crashes 100% of the time in that case. I am using test card `4000 0000 0000 3220`.

```
*** Assertion failure in -[STDSACSNetworkingManager submitChallengeRequest:withCompletion:], STDSACSNetworkingManager.m:58
2021-08-10 14:44:48.779690+0200 Nyx[25941:5172779] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'STDSACSNetworkingManager is not intended to handle multiple concurrent tasks.'
*** First throw call stack:
(0x19ff9f298 0x1b3cf9480 0x19feaccc8 0x1a11e4128 0x106902008 0x10692d51c 0x10692fa2c 0x106910774 0x1a2849ca4 0x1a21d9af4 0x1a21d9e34 0x1a21d8784 0x1a28842dc 0x1a2885bc4 0x1a28617f0 0x1a28e4bcc 0x1a28e93f0 0x1a28e06dc 0x19ff1e8f0 0x19ff1e7f0 0x19ff1db44 0x19ff18060 0x19ff17818 0x1b661d570 0x1a28430e8 0x1a2848664 0x104e2fee0 0x19fbf6140)
libc++abi: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'STDSACSNetworkingManager is not intended to handle multiple concurrent tasks.'
terminating with uncaught exception of type NSException
```

## Code to reproduce
```swift
STPPaymentHandler.shared().handleNextAction(forPayment: someObject.clientSecret, with: self, returnURL: nil) { status, _, error in

    // we don't make it this far
    
}
```

## iOS version
14.6

## Installation method
Pod

## SDK version
(21.7.0)

## Other information

This could probably be resolved by not enabling the cancel button until the VC is ready to handle dismissal.

**Comment by davidme-stripe** [comment]
Thanks for filing this, that's quite bad! We'll deliver a fix in an upcoming SDK update.

**Comment by sfriedman-stripe** [comment]
This was resolved in #2096. Closing
